mm/2s2h/CustomMessage/CustomMessage.cpp

```diff
@@ -0,0 +1,64 @@
+
+#include "CustomMessage.h"
+#include <map>
+
+extern "C" {
+#include "z64.h"
+extern PlayState* gPlayState;
+}
+
+static s32 activeMessageModId = -1;
+static s32 activeMessageTextId = -1;
+
+#define DEFINE_MESSAGE(textId, typePos, msg) { SHIP_TEXT_##textId, typePos, msg },
```

Comment:
Maybe we could do what decomp does and concat the end character via `DEFINE_MESSAGE`. This way we don't have to account for it below/later.
(Granted decomp created a header for all the format chars, which we haven't pulled in yet, so all we can do is put the raw code for now)

```suggestion
#define DEFINE_MESSAGE(textId, typePos, msg) { SHIP_TEXT_##textId, typePos, msg "\xBF"},
```